src/foremast/datapipeline/datapipeline.py

```
@@ -0,0 +1,114 @@
+        self.client = session.client('datapipeline')
+        self.pipeline_id = None
+
+    def create_datapipeline(self):
+        """Creates the data pipeline if it does not already exist"""
+
+        tags = [
+                {"key": "app_group",
+                "value": self.group},
+                {"key": "app_name",
+                "value": self.app_name}
+                ]
+        try:
+            response = self.client.create_pipeline(name=self.datapipeline_data.get('name', self.app_name),
+                                               uniqueId=self.app_name,
+                                               description=self.datapipeline_data['description'],
+                                               tags=tags)
+            self.pipeline_id = response.get('pipelineId')
+        except Exception as e:
+            LOG.warning(e)
+            raise ErrorCreatingDataPipeline
+        
+        LOG.debug(response)
+        LOG.info("Successfully configured Data Pipeline - %s", self.app_name)
+
+    def set_pipeline_definition(self):
+        """Translates the json definition and puts it on created pipeline
+
+        Returns:
+                bool: True if successful
+        """
+
+        if not self.pipeline_id:
+            self.get_pipeline_id()
+
+        json_def = self.datapipeline_data['json_definition']
+        try:
+            pipelineObjects = translator.definition_to_api_objects(json_def)
+            parameterObjects = translator.definition_to_api_parameters(json_def)
+            parameterValues = translator.definition_to_parameter_values(json_def)
+            response = self.client.put_pipeline_definition(
+                                pipelineId=self.pipeline_id,
+                                pipelineObjects=pipelineObjects,
+                                parameterObjects=parameterObjects,
+                                parameterValues=parameterValues)
+        except Exception as e:
```

Don't catch a broad `Exception`. You were handling `awscli.customizations.datapipeline.translator.PipelineDefinitionError` before, so only handle that scenario. Otherwise, we will end up with masked failures.